src/expr/src/relation/mod.rs

```diff
@@ -1962,6 +1962,34 @@ pub struct RowSetFinishing {
     pub project: Vec<usize>,
 }
 
+impl<'a> DisplayText<RenderingContext<'a>> for RowSetFinishing {
```

Comment:
What if this were `DisplayText<Indent>` instead of `DisplayText<RenderingContext>`? We can just pass a mutable reference to the `indent` field of the `RenderingContext`. It will make unit tests easier.

Reply:
That said, my ability to create an unit test is also being impeded by the inability to create a fmt::Formatter.